### ISSUE TITLE
Switch from electron

### ISSUE THREAD

**Issue by thunderstorm010** [BetterDiscord/Installer#238]
So, this is not a problem with the installer but for installing something (I think) we shouldn't need to download some 63.4MB of installer file (on Windows)

Is there any plans for this?

**Comment by tropicaaal** [comment]
At the moment I don't think we have much of an option. I did some testing with [tauri](https://tauri.studio/) and [vite](https://vitejs.dev/guide/) and got a basic app with the UI kit running.

![image](https://user-images.githubusercontent.com/42101043/153498988-d12256a8-2ae2-4bfc-af67-553cdb8c730b.png)

Tauri would significantly reduce bundle size since they don't bundle chromium or node. The problem with tauri is that they don't appear to have an option to allow for portable executables (due to the fact that they utilize WebView2 on windows). This is obviously a problem because we'd need an installer for our installer.

Then we have the native options, which are pretty limited in terms of tooling and out of my area of expertise. The purpose of this installer was to have a single cross-platform portable binary that people could run and install. The platform-specific options (win32/UWP or SwiftUI) are out of the question since we want this to work on every platform from the same codebase. I guess there's GTK and QT which both have their own shortcomings and in my opinion aren't worth the complete rewrite.

Which I suppose leaves React Native and Flutter. Honestly haven't explored these two and I guess they could be options, but to be honest I don't think a complete rewrite of the installer is worth it at that point.

In conclusion, Electron serves a pretty specific set of needs for us at the moment, which is a portable, cross-platform and (more or less) consistent experience across the board and I don't think it's worth the effort (at least at the moment) to warrant a full rewrite in a different framework just to save on a moderate bundle size.
